bin/critterget.py

```diff
@@ -1009,31 +1029,111 @@ def getTimeseriesTrends(appId, appName, trendsData):
     return
 
 
-def getCrashCounts(appId, appName):
-    """Get the number of daily app crashes for a given app.
-    good candidate for 'backfill' data if needed"""
+def get_error_counts(app_id, app_name, error_type):
+    """
+    Get the number of daily handled exceptions for a given app.
 
-    crash_data = apicall("app/crash/counts/{}".format(appId))
+    :param app_id: string for API call
+    :param app_name: string for Splunk
+    :param error_type: string, either 'crash' or 'exception'
+    :return: None
+    """
+    if error_type == 'crash':
+        error_data = apicall("app/crash/counts/{}".format(app_id))
+        error_type_message = 'CrashCounts'
+    elif error_type == 'exception':
+        error_data = apicall("app/exception/counts/{}".format(app_id))
+        error_type_message = 'ExceptionCounts'
 
     print_string = u''
 
     try:
-        for series in crash_data[DATA]:
+        for series in error_data[DATA]:
             print_string += u'({},{}),'.format(series[DATE], series[VALUE])
 
-        print (u'{} MessageType=CrashCounts appName="{}" appId="{}" '
-               u'DATA {}').format(DATETIME_OF_RUN, appName, appId, print_string)
+        print (u'{} MessageType={} appName="{}" appId="{}" DATA {}').format(
+            DATETIME_OF_RUN,
+            error_type_message,
+            app_name,
+            app_id,
+            print_string
+        )
 
     except KeyError as e:
-        print (u'{} MessageType="ApteligentError" Error: Could not access {} '
-               u'in {}.').format(DATETIME_OF_RUN, str(e), 'get_crash_counts')
+        print (u'{} MessageType="ApteligentError" Error: Could not access '
+               u'{} in {}.').format(DATETIME_OF_RUN,
+                                    str(e),
+                                    'get_error_counts')
         return None
 
-    return
+
+def get_error_details(app_id, app_name, error_type):
+    """
+    Get paginated exception summary data and pass it to Splunk
+
+    :param app_id: string
+    :param app_name: string
+    :param error_type: string, either 'crash' or 'exception'
+    :return: None
+    """
+
+    if error_type:
+        error_uri = u'crash/paginatedtable/{}'.format(app_id)
+        error_type_message = 'CrashDetail'
+    else:
+        error_uri = u'exception/paginatedtable/{}'.format(app_id)
+        error_type_message = 'ExceptionDetail'
+
+    exceptions = get_all_pages(error_uri)
```

Comment:
Now that this is generic to errors, we should probably call this 'errors'.  But it's not really a group of errors, is it?  It looks like it's pages maybe?

So this would become something like:


pages = get_all_pages(error_uri)
for page in pages:
  for error in page[DATA][ERRORS]:
    ...